utils/python/CIME/XML/entry_id.py

```diff
@@ -157,12 +157,18 @@ def _set_value(self, node, value, vid=None, subgroup=None, ignore_type=False):
         subgroup is ignored in the general routine and applied in specific methods
         """
         expect(subgroup is None, "Subgroup not supported")
+        valid_values = self.get_optional_node("valid_values", root=node)
         if ignore_type:
             expect(type(value) is str, "Value must be type string if ignore_type is true")
-            node.set("value",value)
+            str_value = value
         else:
             type_str = self._get_type_info(node)
-            node.set("value", convert_to_string(value, type_str, vid))
+            str_value = convert_to_string(value, type_str, vid)
+        if valid_values is not None and valid_values.text is not None and not str_value.startswith('$'):
```

Comment:
I wasn't trying to suggest that the warning should be an error or even that the warning should somehow be removed. I really want to understand the policy on setting XML values to `$var`.
@rljacob , @mvertens , what is the policy on XML values?

1. Anything goes, if the variable is not valid at runtime, the user will get an error then.
2. Only valid XML keys can be used with the `$var` syntax (e.g., `$STOP_OPTION` is okay, `$foo` is not)
3. Environment variables (i.e., not XML keys) are handled with special syntax (e.g., `env(foo)`)
4. Some combination of 2 and 3

I need to understand this policy to correctly evaluate this pull request. Also, we should document the policy.